src/python/pants/backend/python/goals/lockfile.py

```diff
@@ -331,10 +332,27 @@ async def setup_user_lockfile_requests(
     )
 
 
+class PoetryLockfileSentinel(GenerateToolLockfileSentinel):
+    resolve_name = PoetrySubsystem.options_scope
+
+
+@rule
+async def setup_poetry_lockfile(
+    _: PoetryLockfileSentinel, poetry_subsystem: PoetrySubsystem, python_setup: PythonSetup
+) -> GeneratePythonLockfile:
+    # No matter what venv (Python) Poetry lives in, it can still create locks for projects with
```

Comment:
For the record, I went through the exercise of testing this with poetry 1.1.14 installed both in a 2.7 venv and a 3.11 venv creating project locks for the opposite.